library/psa_crypto.c

```diff
@@ -4257,6 +4259,13 @@ static psa_status_t psa_generate_derived_key_internal(
                                             slot->key.data,
                                             slot->key.bytes,
                                             &slot->key.bytes, &bits );
+    /* Enforce a size limit, and in particular ensure that the bit
+     * size fits in its representation type.*/
+    if( bits > PSA_MAX_KEY_BITS )
```

Comment:
This is already tested in `psa_validate_key_attributes` which is called from `psa_start_key_creation` (and tested in `test_suite_psa_crypto` in “PSA key derivation: key too large”). Here, there's only one value of `bits` which is the one from the attributes. Import is different because it can derive `bits` from the data, in which case the validation in `psa_validate_key_attributes` doesn't do anything.